courses/sync_external_courses/utils.py

```diff
@@ -0,0 +1,202 @@
+                "course_runs_without_prices",
+                "External Course Run Codes",
+            ),
+            "course_runs_skipped": StatItemsCollection(
+                "course_runs_skipped",
+                "External Course Run Codes",
+                display_name="Course Runs Skipped due to bad data",
+            ),
+            "course_runs_expired": StatItemsCollection(
+                "course_runs_expired",
+                "External Course Run Codes",
+                display_name="Expired Course Runs",
+            ),
+            "course_runs_deactivated": StatItemsCollection(
+                "course_runs_deactivated",
+                "External Course Run Codes",
+            ),
+            "course_pages_created": StatItemsCollection(
+                "course_pages_created",
+                "External Course Codes",
+            ),
+            "course_pages_updated": StatItemsCollection(
+                "course_pages_updated",
+                "External Course Codes",
+            ),
+            "products_created": StatItemsCollection(
+                "products_created",
+                "Course Run courseware_ids",
+            ),
+            "product_versions_created": StatItemsCollection(
+                "product_versions_created",
+                "Course Run courseware_ids",
+            ),
+            "certificates_created": StatItemsCollection(
+                "certificates_created",
+                "Course Readable IDs",
+                display_name="Certificate Pages Created",
+            ),
+            "certificates_updated": StatItemsCollection(
+                "certificates_updated",
+                "Course Readable IDs",
+                display_name="Certificate Pages Updated",
+            ),
+        }
+
+    def add_stat(self, key, code, title=None, msg=None):
+        """
+        Add an item to a specific stat
+        """
+        if key in self.stats:
+            self.stats[key].add(code, title, msg)
+
+    def add_bulk(self, key, codes):
+        """
+        Add multiple items within the same stat
+        """
+        if key in self.stats:
+            for code in codes:
+                self.add_stat(key, code)
+
+    def remove_duplicates(self, target_stat_key, reference_stat_key):
+        """
+        Filters out items from target stat category whose codes exist in reference stat category.
+        """
+        if target_stat_key not in self.stats or reference_stat_key not in self.stats:
+            return
+
+        self.stats[target_stat_key].difference_update(self.stats[reference_stat_key])
+
+    def log_stats(self, logger):
+        """
+        Log all collected statistics
+        """
+        for stat in self.stats.values():
+            codes = stat.get_codes()
+            logger.log_style_success(f"Number of {stat.display_name}: {len(codes)}.")
+            logger.log_style_success(f"{stat.label}: {codes or 0}\n")
```

Comment:
This is not a generic log method because it won't work for the simple logger.
I would suggest:

- Either make it as generic logging without success styles.
- (Recommended): OR check if we are running it inside a management command and use the Success style logging. Otherwise, use the log.info for tasks etc

Reply:
```
    """
    Detect if the code is executing inside a Django management command.
    """
    return any("manage.py" in arg for arg in sys.argv)
```